trajectorytools/export/to_pandas.py

```diff
@@ -0,0 +1,167 @@
+import collections
+import functools
+import itertools
+from typing import Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+
+from variables import get_variable
+
+
+def melt_df(df: pd.DataFrame, value_name: str, var_name: str):
+    melted_df = pd.melt(
+        df,
+        id_vars=[col for col in df.columns if isinstance(col, str)],
+        value_vars=[col for col in df.columns if not isinstance(col, str)],
+        var_name=var_name,
+        value_name=value_name,
+    )
+    return melted_df
+
+
+def generate_var_df(
+    x: np.ndarray,
+    y: np.ndarray,
+    x_name: str,
+    y_name: str,
+    identities: Optional[Dict[str, List]] = None,
+) -> pd.DataFrame:
+    assert len(x) == y.shape[0]
+    df = pd.DataFrame(
+        data=y.T, columns=list(x)
+    )  # Transpose so that columns are time steps
+    if identities is not None:
+        for key in identities:
+            df[key] = identities[key]
+    return melt_df(df, y_name, x_name)
+
+
+def get_focal_nb_ids(key: str, identities: List):
+    focal_neighbour_ids_conds = list(itertools.product(identities, identities))
+    focals, neighbours = list(zip(*focal_neighbour_ids_conds))
```

Comment:
given that you are coverting to lists immediately, what about using list comprehensions. Much more legible too!

Reply:
sure